app/src/androidTest/java/org/mozilla/fenix/ui/TabbedBrowsingTest.kt

```diff
@@ -215,4 +216,53 @@ class TabbedBrowsingTest {
             }
         }
     }
+
+    @Test
+    fun closePrivateTabTest() {
+        var genericURLS = TestAssetHelper.getGenericAssets(mockWebServer)
+
+        homeScreen {
+        }.togglePrivateBrowsingMode()
+        genericURLS.forEachIndexed { index, element ->
+            navigationToolbar {
+            }.openNewTabAndEnterToBrowser(element.url) {
+                verifyPageContent(element.content)
+            }.openHomeScreen { }
+
+            homeScreen {
+                verifyExistingOpenTabs("Test_Page_${index + 1}")
+                verifyCloseTabsButton("Test_Page_${index + 1}")
+                closeTabViaXButton("Test_Page_${index + 1}")
+                verifySnackBarText("Private tab closed")
+                snackBarButtonClick("UNDO")
+                verifyExistingOpenTabs("Test_Page_${index + 1}")
+                verifyCloseTabsButton("Test_Page_${index + 1}")
+                swipeTabRight("Test_Page_${index + 1}")
+                verifySnackBarText("Private tab closed")
+                snackBarButtonClick("UNDO")
+                verifyExistingOpenTabs("Test_Page_${index + 1}")
+                verifyCloseTabsButton("Test_Page_${index + 1}")
+                swipeTabLeft("Test_Page_${index + 1}")
+                verifySnackBarText("Private tab closed")
+                snackBarButtonClick("UNDO")
+                verifyExistingOpenTabs("Test_Page_${index + 1}")
+                verifyCloseTabsButton("Test_Page_${index + 1}")
+            }
+        }
+    }
+
+    @Test
+    fun closePrivateTabsNotificationTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        homeScreen {
+        }.togglePrivateBrowsingMode()
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(defaultWebPage.url) {
+            mDevice.openNotification()
+            verifyPrivateTabsNotification()
```

Comment:
This will probably fail on devices on Firebase that have pre-existing notifications and system notifications and the Firefox notification requires scrolling in the notification tray. I ran into the same problem with download notifications but haven't found a solution that can scroll the notification tray. Keep an eye on this one as it might not be feasible for now.

Reply:
I've run the checks a few times to see how it behaves on Firebase, so far so good. But I'll keep investigating a method to scroll the notifications tray in case this happens.